File: src/service/ModelBuilder.Models/Project.cs
using System;
using System.Collections.Generic;

namespace Mb.Models
{
    public class Project
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();
        public string Name { get; set; }
        public string Description { get; set; }
        public string ProjectOwner { get; set; }
        public DateTime? LastEdited { get; set; }
        public virtual ICollection<Node> Nodes { get; set; }
        public virtual ICollection<Edge> Edges { get; set; }
    }
}

File: src/client/src/models/application/Project.ts
import Edge from "./Edge";
import Node from "./Node";

interface Project {
  id: string;
  iri: string;
  domain: string;
  version: string;
  name: string;
  isSubProject: boolean;
  description: string;
  projectOwner: string;
  updatedBy: string;
  update: Date;
  nodes: Node[];
  edges: Edge[];
}

export default Project;
